mne/tests/test_dipole.py

```diff
@@ -151,3 +154,57 @@ def test_len_index_dipoles():
 
 
 run_tests_if_main(False)
+
+
+@testing.requires_testing_data
+def test_min_distance_fit_dipole():
+
+    data_path = testing.data_path()
+    raw_fname = data_path + '/MEG/sample/sample_audvis_trunc_raw.fif'
+
+    subjects_dir = op.join(data_path, 'subjects')
+    fname_cov = op.join(data_path, 'MEG', 'sample', 'sample_audvis-cov.fif')
+    fname_trans = op.join(data_path, 'MEG', 'sample',
+                          'sample_audvis_trunc-trans.fif')
+    fname_bem = op.join(subjects_dir, 'sample', 'bem',
+                        'sample-1280-1280-1280-bem-sol.fif')
+
+    raw = Raw(raw_fname, preload=True)
+
+    # select eeg data
+    picks = pick_types(raw.info, meg=False, eeg=True, exclude='bads')
+    info = pick_info(raw.info, picks)
+
+    # Let's use cov = Identity
+    cov = read_cov(fname_cov)
+    cov['data'] = np.eye(cov['data'].shape[0])
+
+    # Simulated scal map
+    simulated_scalp_map = np.zeros(picks.shape[0])
+    simulated_scalp_map[27] = 1
+    simulated_scalp_map[28] = 1
+    simulated_scalp_map[29] = 1
+    simulated_scalp_map[30] = 1
+    simulated_scalp_map[31] = 1
+    simulated_scalp_map[32] = 1
+    simulated_scalp_map[33] = 1
+
+    simulated_scalp_map = simulated_scalp_map[:, None]
+
+    min_distance = 5  # distance in mm
+
+    evoked = EvokedArray(simulated_scalp_map, info, tmin=0)
+
+    info['projs'] = []  # get rid of SSP projections
+    if 'eeg' in evoked:
+        evoked = evoked.copy()
+        avg_proj = make_eeg_average_ref_proj(evoked.info)
```

Comment:
the addition of the average reference should be done automatically in the fit_dipole like we do for the MNE code.